src/index.js

```diff
@@ -1,58 +1,34 @@
 'use strict'
 
-const pull = require('pull-stream/pull')
-const Connection = require('interface-connection').Connection
 const assert = require('assert')
-const PeerInfo = require('peer-info')
 const debug = require('debug')
-const once = require('once')
 const log = debug('libp2p:secio')
 log.error = debug('libp2p:secio:error')
 
 const handshake = require('./handshake')
 const State = require('./state')
+const Wrap = require('it-pb-rpc')
 
-module.exports = {
-  tag: '/secio/1.0.0',
-  encrypt (localId, conn, remoteId, callback) {
-    assert(localId, 'no local private key provided')
-    assert(conn, 'no connection for the handshake  provided')
-
-    if (typeof remoteId === 'function') {
-      callback = remoteId
-      remoteId = undefined
-    }
-
-    callback = once(callback || function (err) {
-      if (err) { log.error(err) }
-    })
-
-    const timeout = 60 * 1000 * 5
-
-    const state = new State(localId, remoteId, timeout, callback)
-
-    function finish (err) {
-      if (err) { return callback(err) }
+async function secure (localPeer, duplex, remotePeer) { // returns duplex
+  assert(localPeer, 'no local private key provided')
+  assert(duplex, 'no connection for the handshake provided')
 
-      conn.getPeerInfo((err, peerInfo) => {
-        encryptedConnection.setInnerConn(new Connection(state.secure, conn))
+  const timeout = 60 * 1000 * 5
```

Comment:
Wow, this is a 5 minute timeout?! I know it's unchanged from master, but this is going to end up exceeding the timeout of the entire connection upgrade, now that the libp2p dialer takes the entire connection upgrade into account. We can probably leave it for now, but unless someone specifically increases their dial timeouts this won't ever get triggered as the connection will get aborted long before.

Reply:
Actually, let's just take this out. The default timeout of `State` is 60s, we should probably just use that.

Reply:
As mentioned in the PR, this is no longer getting used. We can remove it entierly